src/inference_profiler.cc

```diff
@@ -1420,7 +1415,8 @@ InferenceProfiler::CollectData(
     PerfStatus& summary, uint64_t window_start_ns, uint64_t window_end_ns,
     std::vector<RequestRecord>&& request_records)
 {
-  InferenceLoadMode id{summary.concurrency, summary.request_rate};
+  ProfileDataCollector::InferenceLoadMode id{
```

Comment:
I made a new enum called `InferenceLoadMode` to hold the mode for when parsing the CLI options happens, but it turns out this identifier was previously used.

The `ProfileDataCollector` class used this for work regarding writing recorded requests and responses to the profile export file.

I just moved that struct with the same name into the `ProfileDataCollector` class since it's related to that.